packages/opencensus-propagation-jaeger/test/test-jaeger-format.ts

```diff
@@ -104,6 +144,50 @@ describe('JaegerPropagation', () => {
       jaegerFormat.inject(setter, emptySpanContext);
       assert.deepStrictEqual(jaegerFormat.extract(getter), null);
     });
+
+    it('should inject spancontext with 64Bit traceID', () => {
```

Comment:
Do we have test coverage that `inject()` pads traceId shorter than 16/32? From jaeger format spec you linked "Senders SHOULD generate hex strings of exactly 16 or 32 characters in length"

Reply:
no, I think we dont have test coverage.

and it also seems that `inject` doesnt do any formatting https://github.com/census-instrumentation/opencensus-node/blob/31cfd4ea8cf9490a1f4dc7447d00a9089c0b6877/packages/opencensus-propagation-jaeger/src/jaeger-format.ts#L101

however the sanity checks prevent injection of shorter spanId / traceId lengths other than 16 or 32, see https://github.com/census-instrumentation/opencensus-node/blob/31cfd4ea8cf9490a1f4dc7447d00a9089c0b6877/packages/opencensus-propagation-jaeger/src/jaeger-format.ts#L83-L86

imho its not possible a non-padded `traceId` is even tried to be injected:
- on propagation, the `extract` pads the extracted `traceId` if necessary
- if a new trace is started, the id is generated properly

however, the `spanId` could potentially be unpadded as the extractor doesnt format it https://github.com/census-instrumentation/opencensus-node/blob/31cfd4ea8cf9490a1f4dc7447d00a9089c0b6877/packages/opencensus-propagation-jaeger/src/jaeger-format.ts#L61  and then it would fail injection.

am I right ? and should we fix that ?

cc @julianbei

Reply:
> imho its not possible a non-padded `traceId` is even tried to be injected:
> 
> * on propagation, the `extract` pads the extracted `traceId` if necessary
> * if a new trace is started, the id is generated properly

Makes sense to me and I guess this is covered already.

> however, the `spanId` could potentially be unpadded as the extractor doesnt format it
> ...
> and then it would fail injection.

I think it would be good to add if its not too much effort

Reply:
@aabmass friendly reminder of this PR and the security updates. if there is anything me and my colleagues can support you with, would be happy to help

Reply:
Sorry for the slow follow up, I am planning to do it just haven't got around!